src/main/java/wolox/training/models/Book.java

```diff
@@ -71,6 +73,7 @@ public String getTitle() {
     }
 
     public void setTitle(String title) {
+        Preconditions.checkNotNull(title);
```

Comment:
For strings let's do 
checkArgument(title != null and !title.isEmpty())
Change it in all string setters.